v3-sandbox/prototype-api/transactions.md

```diff
@@ -0,0 +1,99 @@
+# Transactions
+
+This section defines the API for transactions.
+
+```
+namespace transactions
+requires common, keys, client
+
+// Defines the status of a transaction. Since we can have custom transaction types based on custom services in the consensus node we can not use an enum here anymore
+@abstraction
+TransactionStatus {
+  @immutable code:int32 // the status code that should be unique based on the consensus node. 
+}
+
+enum BasicTransactionStatus extends TransactionStatus {
+    OK
+    INVALID_TRANSACTION
+    PAYER_ACCOUNT_NOT_FOUND
+    ...          // other status codes should be defined here TO_BE_DEFINED_IN_FUTURE_VERSIONS
+    GRPC_WEB_PROXY_NOT_SUPPORTED
+}
+
+// Id of a transaction
+@abstraction
+TransactionId {
+  @immutable accountId:AccountId // the account that is the payer of the transaction
+  @immutable validStart:zonedDateTime // the start time of the transaction
+  @immutable @optional nonce:int32 // nonce of an internal transaction
+  
+  string toString() // returns id in format TO_BE_DEFINED_IN_FUTURE_VERSIONS
+  string toStringWithChecksum() // returns id in format TO_BE_DEFINED_IN_FUTURE_VERSIONS
+}
+
+// Helper to allow external signing of transactions
+@abstraction
+TransactionSigner {
+  bytes signTransaction(transactionBytes: bytes); // returns the signature as byte array
+}
```

Comment:
What's the thought behind this? It may make sense to have this functionality as part of `KeyPair` or `PrivateKey` which is what the SDKs already have.